File: demo/examples/hello_world/application.c
////////////////////////////////////////////////////////////////////////////////
//
//  CHIPS-2.0 USER DESIGN
//
//  :Author: Jonathan P Dawson
//  :Date: 17/10/2013
//  :email: chips@jondawson.org.uk
//  :license: MIT
//  :Copyright: Copyright (C) Jonathan P Dawson 2013
//
//  Top level ATLYS design
//
////////////////////////////////////////////////////////////////////////////////

#include <stdio.h>
#include <print.h>

int eth_rx = input("eth_rx");
int eth_tx = output("eth_tx");

unsigned rs232_tx = output("rs232_tx");
int rs232_rx = input("rs232_rx");

int timer = input("timer");
int input_switches = input("switches");
int input_buttons = input("buttons");
int output_leds = output("leds");



void application(){

    int a;

    stdout = rs232_tx;
    stdin = rs232_rx;

    puts("Hello World!\n");
    fputc(0xaa, output_leds);

}

File: demo/examples/hello_world/application.py
from demo.components.server import server
from chips.api.api import *

def application(chip):

    hello_world = Component(
    """
    #include <stdio.h>
    stdout = output("rs232_tx");

    void main(){
        puts("Hello World!\n");
    }
    """, inline = True)

    hello_world(
        chip, 
        inputs = {}, 
        outputs = {"rs232_tx":chip.outputs["output_rs232_tx"]},
        parameters = {})

